lib/cadence/saga/concern.rb

```diff
@@ -9,14 +9,21 @@ def run_saga(&block)
 
         block.call(saga)
 
-        Result.new(true)
+        Result.new(Result::COMPLETED)
       rescue StandardError => error # TODO: is there a need for a specialized error here?
         logger.error("Saga execution aborted: #{error.inspect}")
         logger.debug(error.backtrace.join("\n"))
 
-        saga.compensate
+        if compensable?(error)
+          saga.compensate
+          Result.new(Result::COMPENSATED, error)
+        else
+          Result.new(Result::FAILED, error)
```

Comment:
I think in the case we can just re-raise the error. This will raise at the top workflow level and devs can choose to rescue it if they want

Reply:
Makes sense, reverted back to the original result.